DependencyInjection/EnvVarEncryptedProcessor.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the ekino/data-protection-bundle project.
+ *
+ * (c) Ekino
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Ekino\DataProtectionBundle\DependencyInjection;
+
+use Ekino\DataProtectionBundle\Encryptor\EncryptorInterface;
+use Symfony\Component\DependencyInjection\EnvVarProcessorInterface;
+
+/**
+ * @author Rémi Marseille <remi.marseille@ekino.com>
+ */
+final class EnvVarEncryptedProcessor implements EnvVarProcessorInterface
+{
+    /**
+     * @var EncryptorInterface
+     */
+    private $encryptor;
+
+    /**
+     * EnvVarEncryptedProcessor constructor.
+     *
+     * @param EncryptorInterface $encryptor
+     */
+    public function __construct(EncryptorInterface $encryptor)
+    {
+        $this->encryptor = $encryptor;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getEnv($prefix, $name, \Closure $getEnv)
+    {
+        return $this->encryptor->decrypt(\call_user_func($getEnv, $name));
```

Comment:
```suggestion
        return $this->encryptor->decrypt($getEnv($name));
```

Reply:
Didn't do that at the beginning (just `$getEnv($name)`), but it raises an error with phpstan, and I prefer keep this class in the analysis, so no choice...

Reply:
IMHO, PHPStan is just a tool, we are the experts at the end. Let's choose performance over a green quality check. We can exclude this specific error from this file with the "ignoreErrors" config.

Reply:
Do you have any metrics/benchmark somewhere?

Reply:
https://github.com/symfony/symfony/pull/29309 & https://github.com/fab2s/call_user_func

Reply:
Ok, given PHP 7.1.2 over 100,000 iterations with this use case, talking about ~6ms, I don't think the debate is here :). BTW, please note Symfony caches it in https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/Container.php#L412 , so 6ms when deal with 100,000 concurrent requests.

Reply:
I agree it's totally an useless micro optimisation in this case. But it would still be better with the change + we would follow a good practice + it shows that we follow the language evolution and that we care about micro optimisations.

Reply:
So...see https://github.com/ekino/phpstan-banned-code/pull/11 ... 😬